sql/expression/function/distance.go

```diff
@@ -0,0 +1,185 @@
+// Copyright 2020-2021 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package function
+
+import (
+	"errors"
+	"fmt"
+	"math"
+	"strings"
+
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/expression"
+)
+
+// Distance is a function that returns a point type from a WKT string
+type Distance struct {
+	expression.NaryExpression
+}
+
+var _ sql.FunctionExpression = (*Distance)(nil)
+
+// NewDistance creates a new point expression.
+func NewDistance(args ...sql.Expression) (sql.Expression, error) {
+	if len(args) < 2 || len(args) > 3 {
+		return nil, sql.ErrInvalidArgumentNumber.New("ST_DISTANCE", "2 or 3", len(args))
+	}
+	return &Distance{expression.NaryExpression{ChildExpressions: args}}, nil
+}
+
+// FunctionName implements sql.FunctionExpression
+func (d *Distance) FunctionName() string {
+	return "st_distance"
+}
+
+// Description implements sql.FunctionExpression
+func (d *Distance) Description() string {
+	return "returns the minimum distance between any two geometries."
+}
+
+// Type implements the sql.Expression interface.
+func (d *Distance) Type() sql.Type {
+	return sql.Float64
+}
+
+func (d *Distance) String() string {
+	var args = make([]string, len(d.ChildExpressions))
+	for i, arg := range d.ChildExpressions {
+		args[i] = arg.String()
+	}
+	return fmt.Sprintf("ST_DISTANCE(%s)", strings.Join(args, ","))
+}
+
+// WithChildren implements the Expression interface.
+func (d *Distance) WithChildren(children ...sql.Expression) (sql.Expression, error) {
+	return NewDistance(children...)
+}
+
+// FindGeometryDistance finds the minimum distance between points of any geometry types a and b
```

Comment:
this function doesn't find minimum distance between geometries, it finds _a_ distance between two arbitrary points within each geometry

Reply:
Seems like this works fine right? It's N-squared, compares all points in the first geometries with all points in the second